### PR TITLE
Fix support for Uglifier 3

### PR DESCRIPTION
**Pull request by lucasmazza**
See https://github.com/rails/sprockets/commit/386ab2ba215f828602b6e6bfee9bfc95dc23d574#commitcomment-20292856

### CODE REVIEW

File: lib/sprockets/uglifier_compressor.rb [hunk 1]
```diff
@@ -47,14 +47,24 @@ def call(input)
       if Autoload::Uglifier::VERSION.to_i < 2
```
Comment (matthewd) [inline lib/sprockets/uglifier_compressor.rb]: Given this condition is already here, it seems simpler to combine them into a `case`.

File: lib/sprockets/uglifier_compressor.rb [hunk 2]
```diff
@@ -47,14 +47,24 @@ def call(input)
...
         raise "uglifier 1.x is no longer supported, please upgrade to 2.x"
       end
-      @uglifier ||= Autoload::Uglifier.new(@options.merge({ source_filename: input[:filename] }))
+      uglifier = Autoload::Uglifier.new(@options.merge(input_options(input)))
```
Comment (matthewd) [inline lib/sprockets/uglifier_compressor.rb]: You didn't mention why you're removing the ivar memoization... I guess it was invalidated when we started passing the filename in? Is it slow enough that this should concern us, or was it previously a premature optimization?

Comment (lucasmazza) [inline lib/sprockets/uglifier_compressor.rb]: > I guess it was invalidated when we started passing the filename in?

I believe it so, although the source maps of my current app seems to be working as expected even with the wrong filename being used on different assets.

**Comment by schneems** [comment]
Thanks!

**Pull request merged**